internal/databases/sqlserver/blob/server.go

```diff
@@ -783,3 +791,52 @@ func (bs *Server) validateBlobCompressionEncryption(idx *Index) error {
 	}
 	return nil
 }
+
+func (bs *Server) getCachedReader(idx *Index, s Section) (io.ReadCloser, error) {
+	folder := idx.folder
+	key := folder.GetPath() + s.Path
+	if s.BlockSize > MaxCacheBlockSize {
+		tracelog.DebugLogger.Printf("READ_OBJ: %s %d", key, s.BlockSize)
+		r, err := folder.ReadObject(s.Path)
+		if err != nil {
+			return nil, err
+		}
+		return bs.decompressDecryptIfNeeded(idx, r)
+	}
+	var buf []byte
+	if b, ok := bs.readCache.Get(key); ok {
+		tracelog.DebugLogger.Printf("READ_CACHE: %s %d", key, s.BlockSize)
+		buf = b.([]byte)
+	} else {
+		tracelog.DebugLogger.Printf("READ_OBJ: %s %d", key, s.BlockSize)
+		r, err := folder.ReadObject(s.Path)
+		if err != nil {
+			return nil, err
+		}
+		dr, err := bs.decompressDecryptIfNeeded(idx, r)
+		if err != nil {
+			utility.LoggedClose(r, "failed to close block reader")
+			return nil, err
+		}
+		defer dr.Close()
```

Comment:
NIT: you can add `utility.LoggedClose` to wrap `dr.Close()`

Reply:
fixed